code/obj/flock/cage.dm

```diff
@@ -101,76 +175,135 @@
 		egg.throw_at(target, 12, 3)
 
 	process()
-		// consume any fluid near us
-		var/turf/T = get_turf(src)
-		if(T?.active_liquid)
-			var/obj/fluid/F = T.active_liquid
-			F.group.drain(F, 15, src)
-
 		// process fluids into stuff
 		if(reagents.has_reagent(target_fluid, create_egg_at_fluid))
 			reagents.remove_reagent(target_fluid, create_egg_at_fluid)
 			spawnEgg()
-
+		if(occupant && src.flock)
+			src.flock.updateEnemy(occupant)
 		// process stuff into fluids
 		if(isnull(target))
 			// find a new thing to eat
 			var/list/edibles = list()
 			for(var/obj/O in src.contents)
 				edibles += O
-			if(edibles.len >= 1)
+			if(length(edibles))
 				target = pick(edibles)
-				eating_occupant = 0
+				eating_occupant = FALSE
 				playsound(src, "sound/weapons/nano-blade-1.ogg", 50, 1)
 				if(occupant)
 					boutput(occupant, "<span class='notice'>[src] begins to process [target].</span>")
 			else if(occupant && ishuman(occupant))
 				var/mob/living/carbon/human/H = occupant
-				getHumanPiece(H)
-			else if(occupant)
-				occupant.gib() // sorry buddy but if you're some freaky-deaky cube thing or some other weird living thing we can't be doing with this now
+				getHumanPiece(H) //cut off a human part and add it to contents, set it to target
+			else if(occupant && isrobot(occupant))
+				var/mob/living/silicon/robot/H = occupant
+				getRobotPiece(H) //cut off a robot part and add it to contents, set it to target
+			else if(isliving(occupant))
+				eating_occupant = TRUE
+				var/mob/living/M = occupant
+				target = M //set target to the mob
+			else if(iscritter(occupant))
+				eating_occupant = TRUE
+				var/obj/critter/C = occupant
+				C.CritterDeath() //kill obj/critters immediately because their behaviour is jank and awful
+				target = C //set target to the critter
+
 			if(target)
 				target.set_loc(src)
 		else
-			underlays -= target
 			if(hasvar(target, "health"))
```

Comment:
does this work for mob critters? since they use the health holder system.
If so sure yeah whatever; I'll roll with it for now but we really gotta fix this sometime in the future

Reply:
yeah, it works fine - essentially it treats everything like an item, so if there are limbs and stuff to rip off, they have item health. If all the organs/limbs have been consumed *then* it eats the occupant's body - by this point they're already well dead.

Reply:
Also if critters use a health holder instead, their health var will default to 100 so it continues to work, but may have some odd results if a critter has a very large/very small health holder thing but ignores the health property.

Reply:
gonna write that off as an issue with the health system and not this
👍